src/Dobble/Deck.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Marmelab\Dobble;
+
+/**
+ * @todo generate functions
+ */
+class Deck implements \Countable
+{
+    private $cards = [];
+
+    public function __construct(array $cards = [])
+    {
+        foreach ($cards as $card) {
+            $this->append($card);
+        }
+    }
+
+    public function __toString()
+    {
+        return sprintf('<DobbleDeck: [%s]>', implode(', ', $this->cards));
+    }
+
+    public function append(Card $card)
+    {
+        array_push($this->cards, $card);
+    }
+
+    public function getCards()
+    {
+        return $this->cards;
+    }
+
+    public function count()
+    {
+        return count($this->cards);
+    }
+
+    public function validate()
+    {
+        if (!count($this)) {
+            // The deck is empty
+            return false;
+        }
+
+        if (count(array_unique($this->cards)) < count($this)) {
+            // The deck has two or more identical cards
+            return false;
```

Comment:
I would instead throw exceptions, to give more details on why a deck is not valid.